src/Components/Components/src/RenderTree/RenderTreeDiffBuilder.cs

```diff
@@ -525,6 +525,17 @@ private static object KeyValue(ref RenderTreeFrame frame)
             newComponentFrame.ComponentStateField = componentState;
             newComponentFrame.ComponentIdField = componentState.ComponentId;
 
+            var newParametersLifetime = new ParameterViewLifetime(diffContext.BatchBuilder);
+            var newParameters = new ParameterView(newParametersLifetime, newTree, newComponentIndex);
+            if (renderer.HotReloadContext.IsHotReloading())
+            {
+                // When performing hot reload, we want to force all components.
+                // We do this using two mechanisms - we call SetParametersAsync even if the parameters
+                // are unchanged and we ignore ComponentBase.ShouldRender
+                componentState.SetDirectParameters(newParameters);
+                return;
```

Comment:
Rather than having two separate code paths leading into `componentState.SetDirectParameters`, would it work to change the following below?

```
if (!newParameters.DefinitelyEquals(oldParameters))
```

to

```
if (!newParameters.DefinitelyEquals(oldParameters) || isHotReloading)
```

Ordering the "if" test this way round, even though it's redundant if `isHotReloading` is true, is to optimize for the case where `isHotReloading` is false in production.